### PR TITLE
Add AI processing, store, and search methods to DA [WEB-2934]

### PR DESCRIPTION
**Pull request by web-dev-trev**

### CODE REVIEW

File: app/Console/Commands/BaseCommand.php [hunk 1]
```diff
@@ -67,4 +67,9 @@ protected function getModels()
             })
             ->collapse();
     }
+
+    protected function getAicLogo()
+    {
+        return $this->info(" --------------- \n|               |\n|               |\n|               |\n|               |\n|               |\n|               |\n|        A I C  |\n --------------- ");
```
Comment (zachgarwood) [inline app/Console/Commands/BaseCommand.php:73]: 💖

File: config/azure.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    |
+    | Configuration endpoints and keys for Azure AI Cognitive Services
+    |
+    */
+    'status' => env(key: 'AI_SERVICES_ENABLED'),
+
+    'embedding' => [
+        'endpoint' => env(key: 'AZURE_EMBEDDING_API_ENDPOINT'),
+        'key' => env(key: 'AZURE_EMBEDDING_API_KEY'),
+        'version' => env(key: 'AZURE_EMBEDDING_API_VERSION')
```
Comment (zachgarwood) [inline config/azure.php:18]: Should we set a default for these endpoint `version`s?

**Review by zachgarwood — CHANGES_REQUESTED** [review]
Really interesting!

File: database/migrations/2024_12_02_142650_create_vector_db_and_tables.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+return new class () extends Migration {
+    public function __construct()
+    {
+        $this->connection = 'vectors';
+    }
+
+    public function up(): void
+    {
+        // Ensure extension is enabled on the database
+        DB::statement('CREATE EXTENSION IF NOT EXISTS vector');
```
Comment (nikhiltri) [inline database/migrations/2024_12_02_142650_create_vector_db_and_tables.php:17]: Dupe from previous migration, delete this line.

**Review by nikhiltri — APPROVED** [review]
Good work! ⭐ 

I think this is ready to go. I left a few question in here. Let's review and deploy together during our 1:1.

**Pull request merged**